cardinal/search/search.go

```diff
@@ -167,7 +167,39 @@ func (s *Search) Each(eCtx engine.Context, callback CallbackFn) (err error) {
 	return nil
 }
 
-func (s *Search) collect(eCtx engine.Context) ([]types.EntityID, error) {
+func maxid(ids []types.EntityID) types.EntityID {
+	var acc types.EntityID = 0
+	for _, id := range ids {
+		if id > acc {
+			acc = id
+		}
+	}
+	return acc
+}
+
+func fastSortIds(ids []types.EntityID) {
```

Comment:
we can also just use `slices.Sort` , it uses pdqs under the hood, so should be fast enough

Reply:
https://pkg.go.dev/sort#Sort

The sort here is actually O(N). Technically, O(max_entity_id_value). It works because of the way we create entity ids.

Reply:
I switched this so slices.Sort. I was mistaken about the way we create entity ids. This will cause a problem if we use this style of sort.